pandera/backends/polars/checks.py

```diff
@@ -77,9 +77,7 @@ def apply(self, check_obj: PolarsData):
 
     def postprocess(self, check_obj, check_output):
         """Postprocesses the result of applying the check function."""
-        if isinstance(check_obj, PolarsData) and isinstance(
-            check_output, pl.LazyFrame
-        ):
+        if isinstance(check_output, pl.LazyFrame):
             return self.postprocess_lazyframe_output(check_obj, check_output)
         elif isinstance(check_output, bool):
```

Comment:
@cosmicBboy There was some inconsistency here, as `check_obj` type isn't being checked below. If you think it's necessary, maybe can separately raise some error `if not isinstance(check_obj, PolarsData)`; however, I felt it's not necessary, and didn't include the check for the Ibis backend, either.